src/Filesystem/ComposerReader.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the TYPO3 project  - inspiring people to share!
+ * (c) 2020 Oliver Bartsch & Benni Mack
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace TYPO3\Tailor\Filesystem;
+
+/**
+ * Reading information from composer.json
+ */
+class ComposerReader
+{
+    /** @var array */
+    protected $composerSchema = [];
+
+    public function __construct(string $path = '')
+    {
+        $filename = rtrim($path ?: (string)(getcwd() ?: '.'), '/') . '/composer.json';
+        if (!file_exists($filename)) {
+            return;
+        }
+        $content = @file_get_contents($filename);
+        if ($content === false) {
+            return;
+        }
+        $this->composerSchema = json_decode($content, true);
+        if (!$this->composerSchema || $this->composerSchema === []) {
+            throw new \InvalidArgumentException('The composer.json found is invalid!', 1610442954);
```

Comment:
this should crash the application? wouldn't it make sense to throw a specific exception here, catch it somewhere to ignore that file? or is a working composer.json mandatory for us?

Reply:
If a project features a `composer.json`, it should be a valid one. We will add more automatic detection in the future. Therefore, we have to ensure we deal with a valid file here. I however added a dedicated exception which we can then catch if needed.